app/models/spree/order.rb

```diff
@@ -743,5 +745,22 @@ def set_payment_amount!
 
       pending_payments.first.update_attribute :amount, total
     end
+
+    def save_default_addresses
```

Comment:
What shall we do about the complexity? The method could be split like this:

```
(copy_bill_address || copy_ship_address) && user&.save
```

The two methods then include a `return false` in the guard clause. Just an idea.

Reply:
I'll take a look in the morning. The main thing I was trying to avoid was the potential of making four updates (two each on the customer and the user) instead of a maximum of two (one each).

Reply:
Yes, let's still minimise queries. That should be possible though.

Reply:
Alright, I broke it up a bit into smaller/clearer methods and then moved it to a service as well. The `Order` class needs more LOC like I need a hole in the head... :sweat_smile:

Reply:
Thanks for reviewing! :heart: